Library/Homebrew/formula.rb

```diff
@@ -957,8 +957,14 @@ def unlock
   end
 
   def migration_needed?
-    oldname && !rack.exist? && (dir = HOMEBREW_CELLAR/oldname).directory? &&
-      !dir.subdirs.empty? && tap == Tab.for_keg(dir.subdirs.first).tap
+    return false unless oldname
+    return false if rack.exist?
+
+    old_cellar_dir = HOMEBREW_CELLAR/oldname
```

Comment:
I think you can rename `old_cellar_dir` to `old_rack` because that's what it is.